Source/FuseJS/Environment.uno

```diff
@@ -64,13 +66,70 @@ namespace FuseJS
 			AddMember(new NativeProperty<bool, bool>("dotnet", defined(DOTNET))); // Undocumented for testing use only
 			AddMember(new NativeProperty<bool, bool>("host_mac", defined(HOST_MAC))); // Undocumented for testing use only
 			AddMember(new NativeProperty<string, string>("mobileOSVersion", GetMobileOSVersion));
+			AddMember(new NativeProperty< string, object >("locale", GetCurrentLocale));
 		}
-		
+
 		static string GetMobileOSVersion()
 		{
 			if defined(iOS) return Fuse.iOSDevice.OperatingSystemVersion.ToString();
 			if defined(Android) return Fuse.AndroidProperties.ReleaseVersion;
 			return "";
 		}
+
+		[Foreign(Language.Java)]
+		static extern(Android) string GetCurrentLocale()
+		@{
+			java.util.Locale loc = java.util.Locale.getDefault();
+
+			final char separator = '-';
+			String language = loc.getLanguage();
+			String region = loc.getCountry();
+			String variant = loc.getVariant();
+
+			if (language.isEmpty() || !language.matches("\\p{Alpha}{2,8}")) {
+				language = "und"; // "und" for Undetermined
+			} else if (language.equals("in")) {
+				language = "id";  // correct deprecated "Indonesian"
+			}
+
+			StringBuilder bcp47Tag = new StringBuilder(language);
+			if (!region.isEmpty()) {
+				bcp47Tag.append(separator).append(region);
+			}
+
+			if (!variant.isEmpty()) {
+				bcp47Tag.append(separator).append(variant);
+			}
+
+			return bcp47Tag.toString();
+		@}
+
+		[Foreign(Language.ObjC)]
+		static extern(iOS) string GetCurrentLocale()
+		@{
+			NSString* language = NSLocale.preferredLanguages[0];
+
+			if (language.length <= 2) {
+				NSLocale* locale = NSLocale.currentLocale;
+				NSString* localeId = locale.localeIdentifier;
+				NSRange underscoreIndex = [localeId rangeOfString: @"_" options: NSBackwardsSearch];
+				NSRange atSignIndex = [localeId rangeOfString: @"@"];
+
+				if (underscoreIndex.location != NSNotFound) {
+					if (atSignIndex.length == 0)
+						language = [NSString stringWithFormat: @"%@%@", language, [localeId substringFromIndex: underscoreIndex.location]];
+					else {
+						NSRange localeRange = NSMakeRange(underscoreIndex.location, atSignIndex.location - underscoreIndex.location);
+						language = [NSString stringWithFormat: @"%@%@", language, [localeId substringWithRange: localeRange]];
+					}
+				}
+			}
+
+			return [language stringByReplacingOccurrencesOfString: @"_" withString: @"-"];
+		@}
+
+		static extern(!Mobile) string GetCurrentLocale() {
+			return "en-EN";
```

Comment:
Are you sure `en-EN` is a valid locale? `en-US` sounds more familiar to me.

Reply:
Fixed